include/pybind11/stl_bind.h

```diff
@@ -19,73 +19,87 @@ PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
-struct is_comparable : std::false_type { };
+struct is_comparable : std::false_type {};
 
 /* For non-map data structures, check whether operator== can be instantiated */
 template <typename T>
 struct is_comparable<
-    T, enable_if_t<container_traits<T>::is_element &&
-                   container_traits<T>::is_comparable>>
-    : std::true_type { };
+    T,
+    enable_if_t<container_traits<T>::is_element && container_traits<T>::is_comparable>>
+    : std::true_type {};
 
-/* For a vector/map data structure, recursively check the value type (which is std::pair for maps) */
+/* For a vector/map data structure, recursively check the value type (which is std::pair for maps)
+ */
 template <typename T>
 struct is_comparable<T, enable_if_t<container_traits<T>::is_vector>> {
-    static constexpr const bool value =
-        is_comparable<typename T::value_type>::value;
+    static constexpr const bool value = is_comparable<typename T::value_type>::value;
 };
 
 /* For pairs, recursively check the two data types */
 template <typename T>
 struct is_comparable<T, enable_if_t<container_traits<T>::is_pair>> {
-    static constexpr const bool value =
-        is_comparable<typename T::first_type>::value &&
-        is_comparable<typename T::second_type>::value;
+    static constexpr const bool value = is_comparable<typename T::first_type>::value
+                                        && is_comparable<typename T::second_type>::value;
 };
 
 /* Fallback functions */
-template <typename, typename, typename... Args> void vector_if_copy_constructible(const Args &...) { }
-template <typename, typename, typename... Args> void vector_if_equal_operator(const Args &...) { }
-template <typename, typename, typename... Args> void vector_if_insertion_operator(const Args &...) { }
-template <typename, typename, typename... Args> void vector_modifiers(const Args &...) { }
+template <typename, typename, typename... Args>
+void vector_if_copy_constructible(const Args &...) {}
+template <typename, typename, typename... Args>
+void vector_if_equal_operator(const Args &...) {}
+template <typename, typename, typename... Args>
+void vector_if_insertion_operator(const Args &...) {}
+template <typename, typename, typename... Args>
+void vector_modifiers(const Args &...) {}
 
-template<typename Vector, typename Class_>
+template <typename Vector, typename Class_>
 void vector_if_copy_constructible(enable_if_t<is_copy_constructible<Vector>::value, Class_> &cl) {
     cl.def(init<const Vector &>(), "Copy constructor");
 }
 
-template<typename Vector, typename Class_>
+template <typename Vector, typename Class_>
 void vector_if_equal_operator(enable_if_t<is_comparable<Vector>::value, Class_> &cl) {
     using T = typename Vector::value_type;
 
     cl.def(self == self);
     cl.def(self != self);
 
-    cl.def("count",
-        [](const Vector &v, const T &x) {
-            return std::count(v.begin(), v.end(), x);
-        },
+    cl.def(
+        "count",
```

Comment:
Ugh? But but but ... "def count:" is nice if it's on one line, no? :-(

Reply:
You can play with "priorities" of breaks and have a fine grained control over where clang-format inserts new lines. I played with these priorities once for two hours and didn't achieve what I want, but that's a different story.

Reply:
Sure, yes. Just meant as another example where a human might have thought about this, and a formatter doesn't understand the semantics. pybind11's own docs suggest the previous style: https://pybind11.readthedocs.io/en/stable/classes.html#binding-lambda-functions

Reply:
The problem with putting them on one line is what do you do with the next line? For example,

```cpp
my_really_fantastic_and_great_object_of_some_sort.def("count",
                                                      a_lambda_or_somethign_else_1,
                                                      a_lambda_or_somethign_else_2
);
```

Gives you potentially very little space, and most editors won't get the spacing right for you when you are writing it (since it's a variable indent).

```cpp
my_really_fantastic_and_great_object_of_some_sort.def("count",
    a_lambda_or_somethign_else_1,
    a_lambda_or_somethign_else_2
);
```

makes the "count" much harder to notice/read with the other items.

```cpp
my_really_fantastic_and_great_object_of_some_sort.def(
    "count",
    a_lambda_or_somethign_else_1,
    a_lambda_or_somethign_else_2
);
```

is the only one that is consistent and readable if the arguments and/or object name are sometimes lengthy.


There are quite a few packing controls, though. I think I use the experimental auto packing flag somewhere else.

Reply:
Yeah, sure, I can see that. But here it's just `cl.def`, short enough, which might influence judgment?

At any rate, I find it a weird inconsistency with the docs.

Reply:
For what it is worth, I also prefer not breaking before the first argument if there's enough space, like in this case.